### ISSUE TITLE
Local/Remote Path Conversion Bug

### ISSUE THREAD

**Issue by zhona9** [cristianbuse/VBA-FileTools#26]
This is my Diagnostics, I am not having any issues. The correct local path is returned.
[AmeliaDiagnosticsOD.txt](https://github.com/user-attachments/files/16092878/AmeliaDiagnosticsOD.txt)

This the Diagnostics for the user having issues. For her, the local path returned is "".
[YannTyngDiagnosticsOD.txt](https://github.com/user-attachments/files/16092880/YannTyngDiagnosticsOD.txt)

**Comment by cristianbuse** [comment]
@zhona9 

I tried both files and ```GetLocalPath``` works fine with both configurations when I ask for "https://south32.sharepoint.com/sites/ContinuousImprovementProjects-119ExtendMENDchecklisttoMetals/Shared Documents/119 Extend MEND checklist to Metals"

Please see [my comment on Guido's gist](https://gist.github.com/guwidoe/038398b6be1b16c458365716a921814d?permalink_comment_id=5110858#gistcomment-5110858) regarding the weird character.

**Comment by cristianbuse** [comment]
Hi @zhona9 ,

I am not sure you saw the comment I left in Guido's gist last week so I am copying the comment below:

I simply cannot replicate the issue. I used the data in both diagnostics files you've sent and they both work fine. In lack of other ideas I tought that maybe character 160 is the issue although I saw that it literally is at line break.

> m_providers.arr(i) has an array item 1, yet it throws runtime error 1004 "Application-defined or object-defined error". This doesn't happen when I run the code at all. I think after that the path becomes an empty string.

This is super weird. It could be a compilation bug in VBA. What happens if you replace this:
```VBA
        With m_providers.arr(i)
            If StrCompLeft(odWebPath, .webPath, vbTextCompare) = 0 Then
                collMatches.Add i
                If Not .isBusiness Then Exit For
                If .isMain Then mainIndex = .accountIndex
            End If
        End With
```
with this:
```VBA
        If StrCompLeft(odWebPath, m_providers.arr(i).webPath, vbTextCompare) = 0 Then
            collMatches.Add i
            If Not m_providers.arr(i).isBusiness Then Exit For
            If m_providers.arr(i).isMain Then mainIndex = m_providers.arr(i).accountIndex
        End If
```
?

**Comment by zhona9** [comment]
Thanks Cristian, I will schedule a time with the user to test this code. Apologies for the delay.

**Comment by cristianbuse** [comment]
Fixed in e299cecbfe3d44a8da99f6a271f66edc9c6746df
